sinks/datadog/README.md

```diff
@@ -0,0 +1,94 @@
+
+* The tag `veneurlocalonly` is stripped and influences forwarding behavior, as discussed below.
+* The tag `veneurglobalonly` is stripped and influences forwarding behavior, as discussed below.
+
+## Lack of Host Tags for Aggregated Metrics
+
+By definition the hostname is not applicable to *global* metrics that Veneur processes. Note that if you
+do include a hostname tag, Veneur will **not** strip it for you. Veneur will add its own hostname as configured to metrics sent to Datadog.
+
+## Metrics
+
+Enabled if `datadog_api_hostname` and `datadog_api_key` are set to non-empty
+values.
+
+* Counters are converted to second-normalized rates with an interval matching the server's `interval`.
+* Gauges are gauges.
+
+The following tags are mapped to Datadog fields as follows:
+
+* The tag `host` to `hostname`
+* The tag `device` to `device_name`
+
+### Compressed, Chunked POST
+
+Datadog's API is tuned for small POST bodies from lots of hosts since they work on a per-host basis. Also there are limits on the size of the body that
+can be POSTed. As a result Veneur chunks metrics in to smaller bits — governed by `datadog_flush_max_per_body` — and sends them (compressed) concurrently to
+Datadog. This is essential for reasonable performance as Datadog's API seems to be somewhat `O(n)` with the size of the body (which is proportional
+to the number of metrics).
+
+We've found that our hosts generate around 5k metrics and have reasonable performance, so in our case 5k is used as the `datadog_flush_max_per_body`.
+
+## Spans
+
+Enabled if `datadog_trace_api_address` and `datadog_api_key` are set to non-empty
+values.
+
+Spans are sent to [Datadog APM](https://www.datadoghq.com/apm/). The following
+rules manage how [SSF](https://github.com/stripe/veneur/tree/master/ssf) spans
+and tags are mapped to Datadog [trace fields](https://docs.datadoghq.com/api/#tracing):
+
+* The SSF field `name` is mapped to the trace's `name` field.
+* The tag `resource` is removed and mapped to the trace's `resource` field.
+* The `type` field is currently hardcoded to "web".
+* The SSF field `error` is mapped to the trace's `error` field.
+* Remaining tags are mapped to the trace's `meta` dictionary.
+
+### Span Retention
+
+Veneur allocates a ring buffer of `datadog_span_buffer_size` entries which is flushed
+every `interval`. Being a ring buffer, the oldest spans will be overwritten
+until it `interval` expires and the ring buffer is reset.
+
+## Other
+
+As a side-effect of implementing [DogStatsD](https://docs.datadoghq.com/developers/dogstatsd/)
+Veneur parses both [Service Checks](https://docs.datadoghq.com/api/#service-checks)
+and [Events](https://docs.datadoghq.com/api/#events).
+
+# Configuration
+
+See the various `datadog_*` keys in [example.yaml](https://github.com/stripe/veneur/blob/master/example.yaml) for all available configuration options.
```

Comment:
Once moved up, this should probably list the two ways you can use the sink in (spans and metrics) and how to achieve each configuration.